Introduce metadata about SIGs and markdown generation

joaopgrassi opened the pull request:
Fixes #2697 

## Changes

Very early on draft of an attempt to encode SIG metadata and render it in a markdown file. The goal is to make which SIGs exist and what are their statuses more transparent to all users.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] Links to the prototypes or existing instrumentations (when adding or changing conventions)

thompson-tomo commented:
@svrnm I suspect that this information would be better suited in the community section of the website/repo rather than duplicated into the sem conv repo.

joaopgrassi commented:
> @svrnm I suspect that this information would be better suited in the community section of the website/repo rather than duplicated into the sem conv repo.

No, we specifically discussed and we want to have this as part of semconv. Not everyone knows even the community repo exists, so we want to encode this info here for transparency. Additionally it will serve as bases for some automation we are planning to do around issue/PR triage/management.

thompson-tomo commented:
> Not everyone knows even the community repo exists

Agree that is why I mentioned the website as well.

I am just mindful of creating yet another location/source of info. As such the definition reside in community repo and the otelbot syncs it to repos?

lmolkova commented on sigs.yaml:
```diff
@@ -0,0 +1,17 @@
+    areas:
+      - area:k8s
+      - area:container
+    labels:
+      - accepting_contributions
+      - needs_prototyping
```
do you think we'd have SIG-specific labels?  trying to think of a scenario and only some super-advanced things come up.  So wonder if we can postpone adding those until we need it?

joaopgrassi replied:
Yeah that's true. I wanted to somehow encode that when someone sends a PR adding conventions, but hasn't linked any implementation/prototyping we would flag that automatically. But I guess you are right, that's probably always our expectations so may not need this label.

The idea I had (which now makes less sense) is when conventions are marked stable, the sig may not need prototyping but that's wrong xD

lmolkova commented on SIGS.md:
nit: maybe `sigs` or `SIGs` ?

Also, should we render it like inside the readme? Or at least link from readme?

lmolkova commented on SIGS.md:
```diff
@@ -0,0 +1,6 @@
+<!-- sigs -->
+| Name | Owners | Project | Board | Areas | Status | Directories | Notes |
```
I wonder if we need directories here - I'd be happy to help automating codeowners generation and this is where this info would be necessary.

joaopgrassi replied:
ah right I see. I guess doesn't hurt to get the quick overview on where to find stuff?

lmolkova reviewed (commented):
thanks A LOT for putting it together! This builds the foundation that allows us to automate triage process and be upfront with community wrt contribution process.

joaopgrassi commented on internal/tools/scripts/update-sig-tables.py:
```diff
@@ -0,0 +1,90 @@
+    notes = sig['notes']
+
+    owners = ",<br/>".join(
+        [f"[{owner['name']}](https://github.com/orgs/open-telemetry/teams/{owner['github']})"
```
Done. Now there's a `type` for owner, which can be used to add both team or individual users

trask reviewed (approved):
> This builds the foundation that allows us to automate triage process and be upfront with community wrt contribution process.

👍

cc @danielgblanco for awareness

lmolkova commented on SIGs.md:
```diff
@@ -0,0 +1,33 @@
+semantic conventions project. In particular, the table below highlights
+the owners of each SIG and their current status.
+
+To learn more about SIGs for example, how to kick-start a new one, see the
```
wonder if we can 
- add a warning that this table is under development
- merge this PR 
- divide and conquer filling SIG details

WDYT?

lmolkova reviewed (approved):
LGTM

checking with @open-telemetry/semconv-k8s-approvers if they have any feedback 

Context: we're using k8s SIG as an example here and will add details for other SIGs as well. 
The yaml will be used to automate triage (e.g. close/comment on the PRs for inactive areas), but the automation will come as a follow up.

kamphaus reviewed (commented):
The CODEOWNERS file mentions other approver groups (eg. feature-flags).
Should those be listed as well?

joaopgrassi commented on sigs.yaml:
```diff
@@ -0,0 +1,134 @@
+    owner:
+      - name: "semconv-system-approvers"
+        github: semconv-system-approvers
+        type: team-user
```
I didn't have it it in the initial draft but thought it would be good to make it flexible. If we don't need it, I can remove it again

joaopgrassi commented:
> The CODEOWNERS file mentions other approver groups (eg. feature-flags). Should those be listed as well?

yeah I wasn't sure if I wanted to add the "finished" sigs as part of this PR (would be too big) but I guess I could.

joaopgrassi commented on sigs.yaml:
```diff
@@ -0,0 +1,134 @@
+sigs:
+  - name: "Semantic Conventions: System"
```
Ok, I added the above - but .NET, Mainframe, Logs, Profiling are all missing either the project board or project document in the community repo (or both). I'm not even sure if those were even properly defined. 

I didn't want to leave the project/board fields as optional, to really enforce consistency and well defined process. For now for those I just used `N/A`. Sucks but.. oh well.

lmolkova replied:
yeah, I think we overlooked the lightweight-process (no sig ever) and already stable (SIG does not exist anymore) cases. added some suggestions to the text

lmolkova commented on SIGs.md:
```diff
@@ -0,0 +1,49 @@
+# Special Interest Groups (SIGs)
+
+Semantic conventions in OpenTelemetry are organized into distinct areas,
+with each area being overseen by a Special Interest Group (SIG). A SIG is
```
```suggestion
with some of them being overseen by Special Interest Group (SIG).
```

we actually have full coverage, right?

lmolkova commented on SIGs.md:
```diff
@@ -0,0 +1,49 @@
+relevant domain and OpenTelemetry itself.
+
+SIGs have several responsibilities and goals, but their primary focus is to
+define and guide semantic conventions toward stabilization. They are also tasked
+with the long-term maintenance of these conventions, serving as domain experts
+for pull request (PR) reviews and issue triage.
```
maybe something around these lines?
There is a push in the commuity repo to have smaller, well-scoped projects and permanent SIGs are seen more as an exception than a common scenario. E.g. we had HTTP or JVM  SIG but they had achieve their goals and are dissolved now.

```suggestion
define and guide semantic conventions toward stabilization. SIGs that achieve their goal 
may be dissolved, but the codeowner group stays serving as domain experts
for pull request (PR) reviews and issue triage.
```

lmolkova commented on SIGs.md:
```diff
@@ -0,0 +1,49 @@
+with the long-term maintenance of these conventions, serving as domain experts
+for pull request (PR) reviews and issue triage.
+
+This document provides an overview of all SIGs involved in the
```
```suggestion
This document provides an overview of all SIGs and codeowner groups involved in the
```

lmolkova commented on sigs.yaml:
```diff
@@ -0,0 +1,217 @@
+sigs:
```
nitpicking and we can always rename it later: those are not really sigs, but components/domains. SIGs come and go, we can form a new DB SIG to improve conventions for semantic/vector search it'd have new project description and board, but it would not change the entry here, codeownership group, etc.

danielgblanco replied:
Worth noting, the current `sigs.yml` in `community` now has an extra [field](https://github.com/open-telemetry/community/blob/7771f05d31542a3d1d6c10d8a3323b4a7808a4ee/sigs.yml#L175) called `roadmapProjectIds` which is used to populate https://github.com/orgs/open-telemetry/projects/158. This will represent the current/recent projects being worked on by a given SIG.

Is the expectation that those align with the projects listed here? Will it be the responsibility of the SIGs to keep this (as well as `community` list up to date).

joaopgrassi replied:
> Worth noting, the current `sigs.yml` in `community` now has an extra [field](https://github.com/open-telemetry/community/blob/7771f05d31542a3d1d6c10d8a3323b4a7808a4ee/sigs.yml#L175) called `roadmapProjectIds` which is used to populate [OpenTelemetry Roadmap](https://github.com/orgs/open-telemetry/projects/158). This will represent the current/recent projects being worked on by a given SIG.
> 
> Is the expectation that those align with the projects listed here? Will it be the responsibility of the SIGs to keep this (as well as `community` list up to date).

@danielgblanco same as above - the project comes and goes so I don't think we need to also use the `id` here. We want to refer to the SemConv area and its "current status". For ex, I see the flow:

- area is inactive, in need for stabilization
- community decides to start a new stabilization project (e.g., db, rpc)
- project is created, this yaml here is updated to reflect the new status (link to project is added, labels are changed to `active`, `accepting_contributions` etc)
- project is done, closed, yaml here is updated to inactive again

danielgblanco replied:
Thanks @joaopgrassi. That makes sense. Reading that back, this list then summarises the different areas, and for each of them the current effort to stabilise it, which should map to an OTel "project".

joaopgrassi replied:
We will also be adding somewhere a link to this roadmap board, filtering for items on the semconv repo. That's another PR coming

joaopgrassi commented:
Not sure why the link check fails for the https://github.com/orgs/open-telemetry/projects/45. It works for me all the time 🤔

danielgblanco commented:
> Not sure why the link check fails for the https://github.com/orgs/open-telemetry/projects/45.

I don't think this is related, but that project was closed in https://github.com/open-telemetry/community/pull/2931

joaopgrassi commented:
@lmolkova I renamed the top level property to `areas`. Changed then field with the areas to `labels` and the other to `status`. I think this looks more "future proof" now and detached from SIGs
